lib/osf-components/addon/components/validated-input/base-component.ts

```diff
@@ -30,39 +31,54 @@ export default abstract class BaseValidatedInput extends Component {
     placeholder?: string;
     disabled: boolean = defaultTo(this.disabled, false);
     shouldShowMessages: boolean = defaultTo(this.shouldShowMessages, true);
+    model?: DS.Model;
 
     // Private properties
     @service i18n!: I18n;
 
-    validation?: ResultCollection; // defined in constructor
-    value: any; // defined in constructor
+    // defined in constructor
+    errors?: string[];
+    value: any;
+    isInvalid?: boolean;
+    isValidating?: boolean;
+    validation?: ResultCollection;
 
-    @computed('validation.options')
+    @computed('errors', 'validation.options')
     get isRequired(): boolean {
-        if (!this.validation) {
+        if (this.changeset && this.errors) {
+            for (const error of this.errors) {
+                const re = /can't be blank/;
+                if (re.exec(error)) {
+                    return true;
+                }
+            }
```

Comment:
See if there is a better way to do this.

Reply:
There was, indeed. A _much_ better way.